### PR TITLE
Add dynamicGroupMetadata in cloud_identity_group

### PR DESCRIPTION
**Pull request by sqin2019**
Add dynamicGroupMetadata in cloud_identity_group

cann't test the change as I don't have a test cloud identity account.

```release-note:enhancement
cloudidentity: added support for `dynamic_group_metadata` block to `google_cloud_identity_group` resource
```

### CODE REVIEW

**Comment by sqin2019** [comment]
Failed with Error 
```
Error: Error creating Group: googleapi: Error 403: Error(3006): This feature (Dynamic Groups) requires a premium SKU.
        
          with google_cloud_identity_group.cloud_identity_groups_dynamic,
          on terraform_plugin_test.tf line 2, in resource "google_cloud_identity_group" "cloud_identity_groups_dynamic":
           2: resource "google_cloud_identity_group" "cloud_identity_groups_dynamic" {
```

File: mmv1/third_party/terraform/services/cloudidentity/resource_cloud_identity_group_test.go.erb [hunk 1]
```diff
@@ -177,3 +185,87 @@ func testAccCheckCloudIdentityGroupDestroyProducer(t *testing.T) func(s *terrafo
...
+resource "google_cloud_identity_group" "cloud_identity_groups_dynamic" {
+  display_name         = "tf-test-my-identity-dynamic-group%{random_suffix}"
+
+  parent = "customers/%{cust_id}"
+
+  group_key {
+    id = "tf-test-my-identity-dynamic-group%{random_suffix}@%{org_domain}"
+  }
+
+  labels = {
```
Comment (slevenick) [inline mmv1/third_party/terraform/services/cloudidentity/resource_cloud_identity_group_test.go.erb:259]: Looks like the automatically added dynamic group label is causing problems here. We may need to either add a DiffSuppressFunc for that label, or add it explicitly to the tests

Comment (sqin2019) [reply mmv1/third_party/terraform/services/cloudidentity/resource_cloud_identity_group_test.go.erb:259]: Thanks, added it to the tests.

**Review by slevenick — CHANGES_REQUESTED** [review]
I'm still seeing this show up in the tests:
              ~ labels                = {
                  - "cloudidentity.googleapis.com/groups.dynamic"          = "" -> null
                    # (1 unchanged element hidden)
                }

Are you able to successfully run the tests?

**Comment by sqin2019** [comment]
The test failed at group creation, added a diffsuppressfunc instead, [workaround](https://github.com/hashicorp/terraform-plugin-sdk/issues/477) to use diffsupressfunc for map.

**Review by slevenick — CHANGES_REQUESTED** [review]
Test failing, please fix

**Pull request closed without merging**